lib/rules/require-super-in-lifecycle-hooks.js

```diff
@@ -157,9 +178,22 @@ module.exports = {
       const body = isNativeSuper ? node.value.body : node.body;
       const hasSuper = hasMatchingNode(body, (bodyChild) =>
         isNativeClass ? isNativeSuper(bodyChild, hookName) : isClassicSuper(bodyChild)
-      );
-      if (!hasSuper) {
-        report(node, isNativeClass, hookName);
+      ).foundMatch;
+
+      if (hookName === 'init' && hasSuper && requireArgs) {
+        const hasSuperWithArguments = hasMatchingNode(
```

Comment:
Can we avoid searching for the node again? We should already have it from the above `hasMatchingNode ` call right, so we can just check if that node has arguments.